### DISCUSSION TITLE
Idea for better handling of osc-52 (clipboard) over mosh

### DISCUSSION THREAD

**Discussion by rrgeorge** [blinksh/blink#1609] [Ideas]
When using OSC-52 over mosh, it creates a problem where every time the session is loaded, (changing tabs, return to blink, etc) the system clipboard is overwritten with whatever osc-52 was last sent from that session. It could be new, or it could be super old. It’s such a nuisance that I stopped using osc-52 in my sessions. But it’s such a useful thing if it worked in a more user friendly way. So onto my idea:
Instead of immediately loading any osc-52 sequence into the system clipboard, have a “terminal clipboard” unique to that terminal that can be displayed with keyboard shortcut that you can manually copy from. You can further that with having the contents auto selected, so you could do some thing like:
Cmd+C -> Display Clipboard, Cmd+C (again) copy it to the system clipboard

**Comment by carloscabanero** [comment]
Thanks! Yeah, this sounds like a good way to handle this and also opens the door to other interesting usages with the clipboard.
